sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/WritableColumnVector.java

```diff
@@ -611,6 +611,9 @@ public final int appendByteArray(byte[] value, int offset, int length) {
 
   public final int appendArray(int length) {
     reserve(elementsAppended + 1);
+    for (WritableColumnVector childColumn : childColumns) {
+      childColumn.reserve(childColumn.elementsAppended + length);
+    }
```

Comment:
`arrayData().reserve(arrayData().elementsAppended + length)`?

Reply:
I don't think this would work for `Map`, as in that case there are 2 child column vectors but `appendArray()` is still used. Alternatively I could fix this by putting these reserves in `ArrayConverter` and `MapConverter`. Then we could manually unroll those loops there, but code that uses `appendArray` would have to be careful to call reserve on the child vectors before calling `appendArray`. WDYT?

Reply:
Ah, I see. I noticed now that the map/array converters share this same method.